File: CU_AlgorithmDesign/ex00e5/file.cpp
#include <iostream>
#include <string>
#include <algorithm>

using namespace std;

void result(int x, int y, string ans, int cons, int mem, int mx)
{
    if (mem < x)
    {
        result(x, y, ans + to_string(0), 0, mem + 1, max(mx, cons));
        result(x, y, ans + to_string(1), cons + 1, mem + 1, mx);
    }
    else
    {
        if (max(mx, cons) >= y)
        {
            // cout << "------" << cons << "-------\n";
            cout << ans << "\n";
        }
        return;
    }
}

int main()
{
    int x, y;
    cin >> x >> y;
    result(x, y, "", 0, 0, false);
}

File: CU_AlgorithmDesign/a64_q2_two_mcs/file.py
def f(l):
	s=0
	al=[0]
	for i in l:
		s+=i
		if s<0:s=0
		al.append(max(al[-1],s))
	return al[1:]
def main():
	n=int(input())
	l=list(map(int,input().split()))
	ans=max(l)
	if ans<=0:return ans
	l,r=f(l),f(l[::-1])
	for i in range(n-1):ans=max(ans,l[i]+r[-i-2])
	return ans
print(main())